### PR TITLE
Fix automation crash

### LINKED ISSUE #6711: Comboboxes in SongEditor allow automation but crash
### Bug Summary
Comboboxes for zooming and snapping in SongEditor allow automation, but when they are drag to an automation track LMMS crashes.

#### Steps to reproduce
Drag zoom combobox arrow in SongEditor to an automation track

#### Expected behavior
I don't see why those controllers (that are gui controllers) should be automatable, but if they are, they shouldn't cause a crash.
Similar controllers in PianoRoll or AutomationEditor do not cause a crash.

#### Actual behavior
It causes a crash.

#### Screenshot
DrowDown-Automate-Crash

#### Affected LMMS versions
1.2.2 and master

Hint:
As far as I found, it happens because in model.cpp we try to get the parentModel()->fullDisplayName(), and those controls' parents do not have that property or something. The string is not null and is not empty, but when accessing to it, it generates an access violation.

### PR DESCRIPTION
**Pull request by michaelgregorius**
Fix for the crash that occurs when the zooming or snapping model of the Song Editor is used for automation.

Resolves #6711

### CODE REVIEW

**Review by lukas-w — APPROVED** [review]

**Comment by superpaik** [comment]
Coding style: I think you should remove underscore on function parameters.
Other than that it works ok.

**Comment by michaelgregorius** [comment]
Underscores and whitespace has been removed with commit fda938fc.

I have also simplified the logic of `Model::fullDisplayName` with commit 9fe7fbd6. This might also have made more obvious a potential problem if the parent has a non-empty name "xyz" but the child has an empty one. This will result in the name "xyz >" being returned.

File: include/Model.h [hunk 1]
```diff
@@ -37,35 +37,18 @@ class LMMS_EXPORT Model : public QObject
...
 public:
-	Model( Model * _parent, QString _display_name = QString(),
-					bool _default_constructed = false ) :
-		QObject( _parent ),
-		m_displayName( _display_name ),
-		m_defaultConstructed( _default_constructed )
-	{
-	}
+	Model(Model * parent, QString displayName = QString(),
+		  bool defaultConstructed = false );
```
Comment (messmerd) [inline include/Model.h:41]:
```suggestion
	Model(Model* parent, QString displayName = QString(),
		  bool defaultConstructed = false);
```

File: src/core/Model.cpp [hunk 2]
```diff
@@ -27,22 +27,47 @@
 namespace lmms
 {
 
+Model::Model(Model * parent, QString displayName, bool defaultConstructed) :
```
Comment (messmerd) [inline src/core/Model.cpp:30]:
```suggestion
Model::Model(Model* parent, QString displayName, bool defaultConstructed) :
```

File: src/core/Model.cpp [hunk 3]
```diff
@@ -27,22 +27,47 @@
...
+}
+
+bool Model::isDefaultConstructed() const
+{
+	return m_defaultConstructed;
+}
+
+Model* Model::parentModel() const
+{
+	return dynamic_cast<Model *>(parent());
```
Comment (messmerd) [inline src/core/Model.cpp:44]:
```suggestion
	return dynamic_cast<Model*>(parent());
```

File: src/core/Model.cpp [hunk 4]
```diff
@@ -27,22 +27,47 @@
...
+}
+
+QString Model::displayName() const
+{
+	return m_displayName;
+}
+
+void Model::setDisplayName( const QString& displayName )
```
Comment (messmerd) [inline src/core/Model.cpp:52]:
```suggestion
void Model::setDisplayName(const QString& displayName)
```

File: src/core/Model.cpp [hunk 5]
```diff
@@ -27,22 +27,47 @@
...
+	m_displayName = displayName;
+}
+
 QString Model::fullDisplayName() const
 {
-	const QString & n = displayName();
-	if( parentModel() ) 
+	const QString n = displayName();
+
+	if(parentModel())
```
Comment (messmerd) [inline src/core/Model.cpp:61]:
```suggestion
	if (parentModel())
```

**Review by messmerd — APPROVED** [review]
Just some very minor style suggestions.
Everything else looks good to me and it worked when I tested it. I've also confirmed that the logic you simplified is still correct.

**Comment by michaelgregorius** [comment]
Thanks for the code review and double check @messmerd! I have implemented all your proposals with commit 0ebc1894.

**Pull request merged**